integration/map-std/src/unstable.ts

```diff
@@ -86,15 +84,11 @@ export function resolveRequestUrl(url: string, options: { parameters: any, secur
   return url;
 }
 
-// Fetch https://fetch.spec.whatwg.org/#fetch-api
-// Undici types (Node.js implementation) https://github.com/nodejs/undici/blob/main/types/fetch.d.ts
-
 // https://fetch.spec.whatwg.org/#headers-class
 export type HeadersInit = Record<string, string> | [string, string][];
-
-export class Headers implements Iterable<[string, string[]]> {
+export class Headers implements Iterable<[string, string]> {
   private guard: 'immutable' | 'request' | 'request-no-cors' | 'response' | 'none';
-  private headersList: Map<string, string[]>;
+  private headersList: Map<string, string>;
```

Comment:
`.get` doesn't split by anything, it returns the joined header value.

Anyway:
Chromium, Firefox - Present headers as joined strings **except** for Set-Cookie header. Sends all headers as joined in the request.
Node.js - Same as chromium and firefox, but additionally can be inspected by logging the object, which reveals that `Set-Cookie` values are stored separately.
Deno - Same as Node.js **except** sends `Set-Cookie`  as separate headers (not joined in the request). Also doesn't support `getSetCookie()`
Safari - Incorrectly joins `Set-Cookie` values, doesn't support `getSetCookie()`.

Example:
```js
let x = new Headers()
x.append("Set-Cookie", "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
x.append("Set-Cookie", "id=bb; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
x.append("set-cookie", "id=cc; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
x.append("test", "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
x.append("test", "id=bb; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
x.append("Test", "id=cc; Expires=Wed, 21 Oct 2030 07:28:00 GMT")
// in Chromium, Firefox, Node.js, Deno
Array.from(x.entries()) == [
  [ "set-cookie", "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT" ],
  [ "set-cookie", "id=bb; Expires=Wed, 21 Oct 2030 07:28:00 GMT" ],
  [ "set-cookie", "id=cc; Expires=Wed, 21 Oct 2030 07:28:00 GMT" ],
  [
    "test",
    "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT, id=bb; Expires=Wed, 21 Oct 2030 07:28:00 GMT, id=cc; E"... 36 more characters
  ]
]
// in Safari (wrong)
[
  ["set-cookie", "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT, id=b…GMT, id=cc; Expires=Wed, 21 Oct 2030 07:28:00 GMT"],
  ["test", "id=aa; Expires=Wed, 21 Oct 2030 07:28:00 GMT, id=b…GMT, id=cc; Expires=Wed, 21 Oct 2030 07:28:00 GMT"]
]
```

There is also [this table in the spec](https://fetch.spec.whatwg.org/#example-header-list-get-decode-split) with examples which shows that splitting is even more complex, as we need to respect quotes.

Another tidbit, chromium defines multiple headers which are allowed to be coalesced (because their values might contain unquoted commas), but that's an internal implementation: https://github.com/chromium/chromium/blob/4f00f9cb98fb4297fd55b0d09a003b722caddc47/net/http/http_util.cc#L407

We also already had this problem when we were working with headers in one-sdk-js and node wasn't returning headers in the right format. I would much rather let the user decide how they want to send and receive their headers and keep the multimap without any unnecessary joining and splitting. It adds no value for the user and the implementation is hardly that much complex.

Reply:
I did very similar thing you have abpve, but you skipped `.get()`, because it returns invalid result (simply joined with `,`).

Yes MultiMap should be there, it is also important how Headers instance is serialized in `http-call` message and how it is picked by host. 

There is another potantial issue with the implementation I am aware of. Lowecasing header names. I am sure there will be server implementation will not implement by specification case-insensitive.

One thing is for sure, this is real mess, because different implementations does it differently.

Reply:
It's not invalid because it's defined like that by the spec [here](https://fetch.spec.whatwg.org/#concept-header-list-get) (`0x2C 0x20` is ", "). It's basically a bug in the spec, which they attempted to rectify with `getSetCookie`, but there are multiple headers which use comas in the wild.

Reply:
Also I don't think we have to adhere to the spec so much we copy all the bugs 🤷‍♀️ We aren't a web platform, we are a server platform and can afford to expose APIs closer to what Node.js has (which has `.rawHeaders` for returning multi-valued headers)